### PR TITLE
Add section on conflicting superinterfaces

### PR DESCRIPTION
**Pull request by natebosch**
Towards #562

Add a section discussing conflicting superinterfaces to the common
problems document.

The issue discusses the restrictions around both consistent member
signatures and consistent generics. An example is given only for the
latter.

### CODE REVIEW

File: src/content/guides/language/sound-problems.md [hunk 1]
```diff
@@ -494,6 +494,34 @@ var maximumOrNull =
...
+A class cannot implement both `C<A>` and `C<B>`, including indirect
+superinterfaces.
+
+#### Example
+
+In the following code,
+class `C` will have conflicting generic interfaces and it will be impossible to
+define valide overrides for some members.
+
+<?code-excerpt "TODO"?>
```
Comment (natebosch) [inline src/content/guides/language/sound-problems.md:513]: @atsansone - is there a doc describing how to set these up? Do we have any best practices for organization, or naming, or anything else I should be aware of?

Comment (atsansone) [reply src/content/guides/language/sound-problems.md:514]: Yup. Kinda hidden: https://github.com/dart-lang/site-shared/blob/main/doc/code-excerpts.md

File: src/content/guides/language/sound-problems.md [hunk 2]
```diff
@@ -494,6 +494,35 @@ var maximumOrNull =
 
 <hr>
 
+### Conflicting Superinterfaces
+
+A class which `implements` more than one superinterface must be able to
+implement valid overrides for every member.
+Each member with a given name requires compatible signatures across the
+superinterfaces.
```
Comment (atsansone) [inline src/content/guides/language/sound-problems.md:502]: **suggestion:** May want to add something to clarify here.
```suggestion
A class which `implements` more than one superinterface, or member,
must be able to implement valid overrides for every member.
Each member with a given name requires compatible signatures across the
superinterfaces.
```

If this is done elsewhere, you don't need to do it here.

File: src/content/guides/language/sound-problems.md [hunk 3]
```diff
@@ -494,6 +494,35 @@ var maximumOrNull =
...
+
+Superinterfaces must not include conflicting generics.
+A class cannot implement both `C<A>` and `C<B>`, including indirect
```
Comment (atsansone) [inline src/content/guides/language/sound-problems.md:505]: **nitpick:** You can use `can't` here. In fact, [Google Style prefers it](https://developers.google.com/style/contractions#negation-contractions).

File: src/content/guides/language/sound-problems.md [hunk 4]
```diff
@@ -494,6 +494,35 @@ var maximumOrNull =
...
+superinterfaces.
+
+#### Example
+
+In the following code,
+class `C` will have conflicting generic interfaces and it will be impossible to
+define valide overrides for some members.
```
Comment (atsansone) [inline src/content/guides/language/sound-problems.md:512]: **suggestion:** Recommend a little tweak.

```suggestion
class `C` has conflicting generic interfaces.
Define valid overrides for some members would be impossible.
```

**Review by atsansone — APPROVED** [review]
@natebosch : Some suggestions. Great work!

File: src/content/guides/language/sound-problems.md (hunk 2 above)
Comment (natebosch) [reply src/content/guides/language/sound-problems.md:502]: I don't think this is correct - I'll tweak the wording for clarity. I'll change to.

```suggestion
A class which `implements` more than one superinterface must be able to
implement valid overrides for every member of every superinterface.
Each member with a given name requires compatible signatures across the
superinterfaces.
```

**Comment by atsansone** [comment]
@natebosch : The code excerpts need updating.

**Comment by natebosch** [comment]
Sorry for the delay. Coming back around to this, I'm not sure what argument to pass to `dart run excerpter` to validate that I did this correctly. I tried passing it both the `.md` file and the `.dart` file that I edited. In both cases the output is `Processed 0 out of 0 files: 0 out of 0 excerpts visited need to be updated.`

**Comment by natebosch** [comment]
Using `dart run dart_site refresh-excerpt` works for me.

**Pull request merged**